internal/graph/check.go

```diff
@@ -155,7 +156,7 @@ func (cc *ConcurrentChecker) checkInternal(ctx context.Context, req ValidatedChe
 	}
 
 	// Deduplicate any incoming resource IDs.
-	resourceIds := slicez.Unique(req.ResourceIds)
+	resourceIds := lo.Uniq(req.ResourceIds)
```

Comment:
I really wanted to swap this to use `mapz.NewSet[string]`. That would allow loop in this function to be removed
https://github.com/authzed/spicedb/blob/fe18f6700b1104f3f76807128cc0bb12508c2654/internal/graph/check.go#L534

```
	if resourceIds.Has(subject.ObjectId) {
		membershipSet := NewMembershipSet()
		membershipSet.AddDirectMember(subject.ObjectId, nil)
		resourceIds.Remove(subject.ObjectId)
		return membershipSet, resourceIds
	}
```

This blew up and I ended up updating a ton of stuff to use the set structure. This may or may not be a _bad_ idea, but it was getting out of hand enough that I thought I'd ask for thoughts from other before attempting that change. It was putting me down the path of updating [RelationshipsFilter](https://github.com/authzed/spicedb/blob/fe18f6700b1104f3f76807128cc0bb12508c2654/pkg/datastore/datastore.go#L60) to be a `mapz.Set`. This falls outside of the `internal/` package, so I'm not sure if that interface is something that can safely be changed.

Reply:
Meh. Probably not worth it as I think about it.

Reply:
If it reduces memory or CPU, then it might be worth it. You'd have to benchmark though

Reply:
I think it's more understandable API contract to return a set if you're guaranteeing uniqueness.

We should follow the execution path, if there are lots of appends, it's probably less efficient.

Reply:
Alright thanks. I'll play with this idea in a separate PR.